helix-term/src/application.rs

```diff
@@ -845,3 +851,26 @@ impl Application {
         Ok(())
     }
 }
+
+#[cfg(not(windows))]
+fn get_signals() -> Signals {
+    const HX_SIGTSTP: libc::c_int = 20;
+    const HX_SIGCONT: libc::c_int = 18;
+    const HX_SIG_IGN: libc::sighandler_t = 1;
+    const HX_SIG_ERR: libc::sighandler_t = 0;
+    let mut app_signals = vec![HX_SIGTSTP, HX_SIGCONT];
+    let mut enable_sigtstp = true;
+    unsafe {
+        // If SIGTSTP is SIG_IGN, then do not listen for it
+        if libc::signal(HX_SIGTSTP, HX_SIG_IGN) != HX_SIG_ERR {
+            log::debug!("Disabling SIGTSTP, C-z will not suspend Helix");
+            app_signals.remove(0);
+            enable_sigtstp = false;
+        }
+    }
+
+    if let Err(e) = ENABLE_SIGTSTP.set(enable_sigtstp) {
+        eprintln!("ENABLE_SIGTSTP error: {}", e);
+    }
+    Signals::new(app_signals).expect("build signal handler")
```

Comment:
Like `Application::new`, this function should return a `Result` (a `Result<Signals>` I think) and then this line becomes

```suggestion
    Signals::new(app_signals).context("build signal handler")
```

To prevent panicing on building `Signals` (master avoids panicing using this same strategy).